### PR TITLE
port remaining Geomlibs delta from OSF

### PR DESCRIPTION
**Pull request by dassaf4**
Rollup of test enhancements and fixes ported from other geomlibs repos that missed the OSF merge to imodel-native.

Also port some recent bug fixes from PPBase.

Highlights:
- correctly classify coincident arcs in CurveVector::AppendSplitCurvesByRegion
- export periodic B-spline surfaces to IMJS
- export trimmed B-spline surface boundaries to IMJS
- move barycentric functions to DTriangle3d.cpp where they belong
- barycentric functions now report correct signed area
- NURBS "almost equal" logic now compares (normalized) knots
- new test helpers like GTestFileOps::JsonFileToGeometry

### CODE REVIEW

File: iModelCore/GeomLibs/PublicAPI/Geom/DTriangle3d.h [hunk 1]
```diff
@@ -207,7 +194,7 @@ DPoint2dCP pVertex2
 * @param uvw   [out] barycentric coordinates of pPoint relative to T
 * @param dUVWdX   [out] derivatives wrt point.x
 * @param dUVWdY   [out] derivatives wrt point.y
-* @param area  [out] triangle area
+* @param area  [out] signed triangle area
```
Comment (saeeedtorabi) [inline iModelCore/GeomLibs/PublicAPI/Geom/DTriangle3d.h:197]: signed area?! what does it mean?

File: iModelCore/GeomLibs/geom/test/BsplineTest/t_bspBoundary.cpp [hunk 2]
```diff
@@ -1380,1050 +1379,33 @@ TEST(bsptrim, MikeSurface)
...
-    points.insert(points.end(), {DPoint3d::From(0.09807983746729951,0.1193972972110023,-0.10976519872050085), DPoint3d::From(0.09811014639929999,0.11900275355760096,-0.11051833699930103), DPoint3d::From(0.09814045533119861,0.11860820990430148,-0.11127147527809939), DPoint3d::From(0.09817076426310087,0.11821366625100199,-0.11202461355689957), DPoint3d::From(0.09817106735241989,0.118209720814469,-0.11203214493968757), DPoint3d::From(0.0981713704417389,0.118205775377936,-0.11203967632247557), DPoint3d::From(0.09817167353105792,0.118201829941403,-0.11204720770526357), DPoint3d::From(0.0981726848299158,0.11818866545420628,-0.11207233719130576), DPoint3d::From(0.09817369612877368,0.11817550096700956,-0.11209746667734795), DPoint3d::From(0.09817470742763156,0.11816233647981284,-0.11212259616339014)});
-    points.insert(points.end(), {DPoint3d::From(0.034261370422198528,0.11973350499190201,-0.0848945914552005), DPoin
```
Comment (saeeedtorabi) [inline iModelCore/GeomLibs/geom/test/BsplineTest/t_bspBoundary.cpp:2413]: wow! What are all these hard coded numbers?! I see they are used for testing. Why are the removed?

**Review by saeeedtorabi — APPROVED** [review]

File: iModelCore/GeomLibs/geom/test/BsplineTest/t_bspBoundary.cpp (hunk 2 above)
Comment (dassaf4) [reply iModelCore/GeomLibs/geom/test/BsplineTest/t_bspBoundary.cpp:2413]: :) This is an overkill test case that I moved to its own imjs file. I originally created this test before I knew how to read test inputs from a file, lol.

File: iModelCore/GeomLibs/PublicAPI/Geom/DTriangle3d.h [hunk 3]
```diff
@@ -167,18 +167,6 @@ ValidatedTransform LocalToWorldTransform () const;
 ValidatedTransform WorldToLocalTransform () const;
 };
 
-/*----------------------------------------------------------------------------+
```
[the remark this answers is not in the document]
Comment (dassaf4) [reply iModelCore/GeomLibs/PublicAPI/Geom/DTriangle3d.h:170]: Thanks, I meant to keep this one and remove the duplicate block from the .cpp file. I'll restore!

File: iModelCore/GeomLibs/PublicAPI/Geom/DTriangle3d.h (hunk 1 above)
Comment (dassaf4) [reply iModelCore/GeomLibs/PublicAPI/Geom/DTriangle3d.h:197]: The sign comes from the orientation of the triangle and is negative iff the points are clockwise. This number is readily available, so we return it.  (It's half of the denominator in the Cramer's Rule solution to the system.) I'll add an explanation!

**Pull request merged**